browser/src/session.ts

```diff
@@ -0,0 +1,130 @@
+import fs from 'node:fs/promises'
+import os from 'node:os'
+import path from 'node:path'
+import { type IncomingHttpHeaders } from 'node:http'
+import { createHash } from 'node:crypto'
+import { type BrowserContext, type Page } from 'playwright'
+import { newBrowserContext } from './context.ts'
+import { Mutex } from 'async-mutex'
+
+const APP_CACHE_DIR = (() => {
+  const homeDir = os.homedir()
+  const appPath = path.join('otto8', 'tools', 'browser')
+
+  switch (os.platform()) {
+    case 'win32':
+      return path.join(process.env.APPDATA ?? path.join(homeDir, 'AppData', 'Roaming'), appPath)
+    case 'darwin':
+      return path.join(homeDir, 'Library', 'Caches', appPath)
+    default:
+      return path.join(process.env.XDG_CACHE_HOME ?? path.join(homeDir, '.cache'), appPath)
+  }
+})()
+
+let sessionManager: SessionManager | undefined
+
+interface ManagedSession {
+  session: Session
+  cleanupTimeout?: NodeJS.Timeout
+}
+
+export class SessionManager {
+  private readonly sessions = new Map<string, ManagedSession>()
+  private readonly sessionsLock: Mutex = new Mutex()
+
+  private constructor () {
+  }
+
+  static async create (): Promise<SessionManager> {
+    sessionManager ??= new SessionManager()
+    return sessionManager
```

Comment:
Session directories can become orphaned if a daemon with "living" sessions is killed. Should we delete orphaned directories at when the session manager is created to handle that?

Reply:
I think this is a good idea. Orphaned session directories could end up taking up a lot of drive space.

Reply:
done